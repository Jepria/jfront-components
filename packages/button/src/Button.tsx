import React from "react"
import styled from "styled-components"
import nextId from "react-id-generator"

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string
  primary?: boolean
}

const ButtonStyled = styled.button<ButtonProps>`
  box-sizing: border-box;
  cursor: pointer;
  border: ${(props) =>
    `${props.theme.button.borderWidth} ${props.theme.button.borderStyle} ${props.theme.button.borderColor}`};
  margin: ${(props) => props.theme.button.margin};
  padding: ${(props) => props.theme.button.padding};
  font-size: ${(props) => props.theme.fontSize.medium};
  font-family: ${(props) => props.theme.fontFamily};
  font-weight: ${(props) => props.theme.button.fontWeight};
  background: ${(props) =>
    props.primary
      ? props.theme.button.primary.bgColor
      : props.theme.button.bgColor};
  border-color: ${(props) =>
    props.primary
      ? props.theme.button.primary.borderColor
      : props.theme.button.borderColor};
  border-radius: ${(props) => props.theme.button.borderRadius};
  color: ${(props) =>
    props.primary
      ? props.theme.button.primary.color
      : props.theme.button.color};
  webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Safari */
  -khtml-user-select: none; /* Konqueror HTML */
  -moz-user-select: none; /* Old versions of Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently
                              supported by Chrome, Edge, Opera and Firefox */
  &:hover:enabled {
    background: ${(props) =>
      props.primary
        ? props.theme.button.primary.hoverBgColor
        : props.theme.button.hoverBgColor};
    color: ${(props) =>
      props.primary
        ? props.theme.button.primary.hoverColor
        : props.theme.button.hoverColor};
    border: ${(props) =>
      `${props.theme.button.borderWidth} ${props.theme.button.borderStyle} ${
        props.primary
          ? props.theme.button.primary.hoverBorderColor
          : props.theme.button.hoverBorderColor
      }`};
  }
  &:active:enabled {
    background: ${(props) =>
      props.primary
        ? props.theme.button.primary.activeBgColor
        : props.theme.button.activeBgColor};
    color: ${(props) =>
      props.primary
        ? props.theme.button.primary.activeColor
        : props.theme.button.activeColor};
    border: ${(props) =>
      `${props.theme.button.borderWidth} ${props.theme.button.borderStyle} ${
        props.primary
          ? props.theme.button.primary.activeBorderColor
          : props.theme.button.activeBorderColor
      }`};
  }
  &:disabled {
    opacity: 0.5;
    cursor: default;
    background: ${(props) =>
      props.primary
        ? props.theme.button.primary.disabledBgColor
        : props.theme.button.disabledBgColor};
    border: ${(props) =>
      `${props.theme.button.borderWidth} ${props.theme.button.borderStyle} ${
        props.primary
          ? props.theme.button.primary.disabledBorderColor
          : props.theme.button.disabledBorderColor
      }`}
    color: ${(props) =>
      props.primary
        ? props.theme.button.primary.disabledColor
        : props.theme.button.disabledColor};
  }
`

ButtonStyled.defaultProps = {
  primary: false,
  theme: {
    fontSize: {
      medium: "12px",
    },
    fontFamily: "arial, tahoma, verdana, helvetica",
    button: {
      borderWidth: "1px",
      borderStyle: "solid",
      margin: 0,
      fontWeight: 700,
      padding: "12px 24px",
      borderColor: "rgb(153, 187, 232)",
      borderRadius: "5px",
      bgColor: "#fff",
      color: "black",
      hoverBgColor: "rgb(217, 235, 251)",
      hoverColor: "black",
      hoverBorderColor: "rgb(153, 187, 232)",
      activeBgColor: "#b6cbe4",
      activeColor: "black",
      activeBorderColor: "1px solid #b6cbe4",
      primary: {
        borderColor: "rgb(153, 187, 232)",
        borderRadius: "5px",
        bgColor: "rgb(217, 235, 251)",
        color: "black",
        hoverBgColor: "rgba(221, 239, 255, 0.55)",
        hoverColor: "black",
        hoverBorderColor: "rgb(153, 187, 232)",
        activeBgColor: "#b6cbe4",
        activeColor: "black",
        activeBorderColor: "#b6cbe4",
      },
    },
  },
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const htmlId = props.id ? props.id : nextId()
    return (
      <ButtonStyled
        id={htmlId}
        ref={ref}
        className={props.className}
        onClick={props.onClick}
        onDoubleClick={(e) => {
          /** IE fix checkbox double-click issue **/
          if ((document as any).documentMode) {
            e.stopPropagation()
            e.currentTarget.click()
          }
        }}
        {...props}
        disabled={props.disabled}
      >
        {props.children}
      </ButtonStyled>
    )
  },
)
