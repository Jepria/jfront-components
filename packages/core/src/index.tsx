export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
