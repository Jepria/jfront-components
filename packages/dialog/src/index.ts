export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'