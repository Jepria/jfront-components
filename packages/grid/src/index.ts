export * from './common'
export * from './jepgrid'
