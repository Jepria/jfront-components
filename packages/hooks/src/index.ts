export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
