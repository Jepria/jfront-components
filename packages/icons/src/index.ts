export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
