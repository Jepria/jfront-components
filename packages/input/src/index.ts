export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
