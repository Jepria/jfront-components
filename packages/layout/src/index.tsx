import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
