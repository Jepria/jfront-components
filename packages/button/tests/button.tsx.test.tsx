import { fireEvent, render, screen } from "@testing-library/react"
import React from "react"
import { Button } from "../src"

test("Checking for the existence of an element Button", () => {
  expect(render(<Button />))
})

test("Checking Button clicking", () => {
  let ButtonClick = false
  render(
    <Button
      onClick={() => {
        ButtonClick = true
      }}
      aria-label={"label2"}
      style={{
        width: "100px",
        height: "50px",
      }}
      value={"Text"}
    />,
  )
  fireEvent.click(screen.getByLabelText("label2"))
  expect(ButtonClick).toEqual(true)
})

test.skip("Matches snapshot ", () => {
  const { asFragment } = render(<Button />)
  expect(asFragment()).toMatchSnapshot()
})
