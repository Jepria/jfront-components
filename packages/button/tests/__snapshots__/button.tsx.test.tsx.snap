// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm cqDHWp"
    id="id3"
  />
</DocumentFragment>
`;
